Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx

```diff
@@ -32,20 +33,31 @@ type FeaturedToolEntry = {
   output: string
 };
 
-function useFeaturedToolMetadata(): FeaturedToolMetadata | undefined {
+type Result<T> =
+  | { status: 'ok', value: T }
+  | { status: 'error', message: string }
+
+function useFeaturedToolMetadata(): Result<FeaturedToolMetadata> | undefined {
   const communitySiteUrl = useCommunitySiteProjectUrl();
   const [ featuredToolResponseData, setFeaturedToolResponseData ] = useState<FeaturedToolResponseData | undefined>(undefined);
+  const [ featuredToolError, setFeaturedToolError ] = useState<string|undefined>(undefined);
   
   useEffect(() => {
     if (communitySiteUrl != null) {
       (async () => {
-        // FIXME Add basic error-handling 
-        const response = await fetch(`https://${communitySiteUrl}${FEATURED_TOOL_URL_SEGMENT}`, { mode: 'cors' });
-
-        // FIXME Validate this JSON using a Decoder
-        const responseData = await response.json() as FeaturedToolResponseData;
-
-        setFeaturedToolResponseData(responseData);
+        try {
+          const response = await fetch(`https://${communitySiteUrl}${FEATURED_TOOL_URL_SEGMENT}`, { mode: 'cors' });
+          if (response.ok) {
+            const responseData = await response.json() as FeaturedToolResponseData;
```

Comment:
We should carry over this FIXME from the previous implementation, as you did for `WorkshopExercises`:

`// FIXME Validate this JSON using a Decoder`

Reply:
Oops. Good catch!

Reply:
I put the comment back.